### DISCUSSION TITLE
PID not working as expected

### DISCUSSION THREAD

**Discussion by Krish-2014** [LemLib/LemLib#236] [Q&A]
Hi,

I am trying to tune PID. Started with lateral PID tuning first. Was able to derive P value till the bot is oscillating. However, for a given same D value, the distance that it travels keep changing every time(both overshoot and undershoot). Also the bot is not going in a straight direction. Can anyone please suggest if the D needs to be increased or decreased further to drive straight for a given distance?

**Comment by meisZWFLZ** [comment]
What's the code you're running? Are you able to provide a video? How are you checking if its undershooting or overshooting?
Did you make sure to set angular PID constants to 0?
